clientv3/watch.go

```diff
@@ -643,12 +646,7 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 	// lazily send cancel message if events on missing id
 }
 
-func (w *watchGrpcStream) newWatchClient() (pb.Watch_WatchClient, error) {
-	// connect to grpc stream
-	wc, err := w.openWatchClient()
-	if err != nil {
-		return nil, v3rpc.Error(err)
-	}
+func (w *watchGrpcStream) newWatchClient() (ws pb.Watch_WatchClient, err error) {
```

Comment:
@heyitsanthony Does the new case need to go through the resume thing? It can know that there is no pending (pending + resuming == 0) thing, and can skip all resume logic. I think the reasoning behind is that we just do not have to care about it since new and resume can be identical anyway. 

My worry is that this func becomes fat and affect readability. And yes, I hope we could group the resuming logic in the func in a better way. Probably move them to a new function.